mongo/options/autoencryptionoptions.go

```diff
@@ -91,6 +98,43 @@ func (a *AutoEncryptionOptions) SetExtraOptions(extraOpts map[string]interface{}
 	return a
 }
 
+// SetTLSConfig applies custom TLS options to configure connections with each KMS provider.
+func (a *AutoEncryptionOptions) SetTLSConfig(tlsOptsMap map[string]map[string]interface{}) (*AutoEncryptionOptions, error) {
```

Comment:
The body of this function and `ClientEncryptionOptions.SetTLSConfig` seem to be identical. If that's the case, create a common function that both `SetTLSConfig` functions use to prevent the behavior of the two functions from diverging.

Reply:
Got it! Added a helper method to `clientencryptionoptions.go` called `applyTLSOptions()`.